Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,278 @@
+                + languageCode
+            );
+            node.addClass(className);
+            node.setAttribute('data-language-code', languageCode);
+
+            return node;
+        },
+
+        /**
+         * Publishes the cancelDiscover and contentDiscovered events
+         *
+         * @method _publishEvents
+         * @protected
+         */
+        _publishEvents: function () {
+            this.publish(CONFIRM, {
+                bubbles: true,
+                emitFacade: true,
+                preventable: false,
+                defaultFn: this._closeLanguageSelectionBox,
+            });
+            this.publish(CANCEL, {
+                bubbles: true,
+                emitFacade: true,
+                preventable: false,
+                defaultFn: this._closeLanguageSelectionBox,
+            });
+        },
+
+        /**
+         * Updates the event handler for the given event
+         *
+         * @method _syncEventHandler
+         * @protected
+         * @param {String} eventName
+         * @param {Function} oldHandler
+         * @param {Function} newHandler
+         */
+        _syncEventHandler: function (eventName, oldHandler, newHandler) {
+            this.detach(eventName);
+            this.after(eventName, function (e) {
+                if ( newHandler ) {
+                    newHandler(e);
+                }
+                this._resetState();
+            });
+        },
+
+        /**
+         * Resets the state of the languageSelectionBox view
+         *
+         * @method _resetState
+         * @protected
+         */
+        _resetState: function () {
+            this.reset('title');
+            this.reset('confirmHandler');
+            this.reset('cancelHandler');
+            this.reset('existingTranslations');
+        },
+
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                title: this.get('title'),
+                newTranslations: this.get('newTranslations'),
+                existingTranslations: this.get('existingTranslations')
+            }));
+            return this;
+        },
+
+        /**
+         * Closes the languageSelectionBox by firing the languageSelectionBoxClose event
+         *
+         * @protected
+         * @method _closeLanguageSelectionBox
+         */
+        _closeLanguageSelectionBox: function () {
+            this.fire('languageSelectionBoxClose');
+        },
+
+        _getAvailableTranslations: function () {
+            return ['eng-GB', 'pol-PL', 'ger-DE', 'fre-FR', 'nno-NO', 'esl-ES', 
+                'jpn-JP', 'rus-RU', 'swe-SE', 'por-BR', 'cze-CZ', 'ara-SA'];
+        },
+
+        _getNewTranslations: function () {
+            var availableTranslations = this._getAvailableTranslations(),
+                translations = Y.clone(availableTranslations),
```

Comment:
this seems a bit complicated, to ease the code, you should index the list of languages by language code. That way, building the list of potential new language becomes super easy.